packages/taquito/src/operations/types.ts

```diff
@@ -568,6 +568,16 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface StakingParams {
```

Comment:
can consider adding `parameter?: TransactionOperationParameter` will make this interface more versatile

Reply:
The reason I prefer not to add that is: it will be a less elegant abstraction. If we have that, then in wallet API's `stake` method, we have to either:

1. ignore it
2. make it required
3. make it optional

I guess you mean option 3. So what happens if the user sets it, but sets it to the wrong value? We should throw and error. This is less elegant than an API that does not allow the user to make mistakes in the first place.

Reply:
wallet api can take `parameter?` too, that's how we make contractCall. If anything we should validate stake op if being passed params should be unit or it's by default unit

Reply:
I still believe that exposing `parameter` to a user of `stake` function, which is optional, but if provided should only be `{entrypoint:'stake', value:{prim: 'Unit'}}` is not a good DX.

Reply:
@dsawali suggestions?